### ISSUE TITLE
Possible bug with MenuList component substitution

### ISSUE THREAD

**Issue by peeveen** [JedWatson/react-select#5491]
I was trying to create a custom MenuList component that contained an input control.

I found this code, and used it as a basis: https://codesandbox.io/embed/m75wlyx3oy

However, my own code was exhibiting a weird problem: the added input control would often (but not always) need two clicks before it would focus.

I eventually tracked down the source of the problem, though I'm not sure why it happens.

Basically, in that sandbox code, if I change example.js, line 133, from this:

```
MenuList: CustomMenuList,
```

... to this:

```
MenuList: (props) => CustomMenuList(props),
```

... then the problem occurs.

Obviously it's an easy workaround, but I thought you might want it brought to your attention.

**Comment by Rall3n** [comment]
Hello @peeveen,

this is **not** a bug regarding the components feature, but instead a basic issue with React component re-rendering.

The identity of the `MenuList` component changes on re-render because the wrapper function around our custom component is being re-created as well. This causes the `<input>` element to be replaced after the first click, creating the need for a second click.

Need to forward props to your custom component? Add them to your `Select` component and access them using the `selectProps` prop on your custom component (e.g. `onMenuInputFocus` in the example CodeSandbox) instead of creating wrapper functions.

**Comment by peeveen** [comment]
Ah, of course, I should have realised.
Thanks for the explanation.
